helper/app-error.ts

```diff
@@ -1,129 +0,0 @@
-/**
- * This file contains the class that derives from Node.js Error class
- * It should be used for providing custom errors with the proper structure
- */
-
-export type ErrorStatusCode = 400 | 401 | 403 | 404 | 409 | 422 | 429 | 500 | 502 | 503 | 504;
-export type ErrorTypes =
-  | 'BadRequest'
-  | 'Unauthorized'
-  | 'Forbidden'
-  | 'NotFound'
-  | 'Conflict'
-  | 'UnprocessableEntity'
-  | 'TooManyRequests'
-  | 'InternalServerError'
-  | 'BadGateway'
-  | 'ServiceUnavailable'
-  | 'GatewayTimeout';
```

Comment:
How about saving these types as HttpErrorStatusCode and HttpErrorTypes and using it in HttpError and child classes?